OpenRA.Mods.Common/Widgets/ProductionTabsWidget.cs

```diff
@@ -36,17 +36,24 @@ public void Update(IEnumerable<ProductionQueue> allQueues)
 		{
 			var queues = allQueues.Where(q => q.Info.Group == Group).ToList();
 			var tabs = new List<ProductionTab>();
+			var removed = new List<int>();
```

Comment:
You can avoid allocating this list by keeping an `int largestUsedName` that you update when enumerating the tabs, then set NextQueueName to that + 1 after the loop.